cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/CoroutineScopeOwner.kt

```diff
@@ -79,6 +80,40 @@ interface CoroutineScopeOwner {
         }
     }
 
+    /**
+     * Synchronously executes use case and saves it's Deferred. By default all previous
+     * pending executions are canceled, this can be changed by the [cancelPrevious].
+     * This version is used for use cases without initial arguments.
+     *
+     * @return [Result] that encapsulates either a successful result with [Success] or a failed result with [Error]
+     */
+    suspend fun <T : Any?> UseCase<Unit, T>.execute(cancelPrevious: Boolean = true) = execute(Unit, cancelPrevious)
+
+    /**
+     * Synchronously executes use case and saves it's Deferred. By default all previous
+     * pending executions are canceled, this can be changed by the [cancelPrevious].
+     * This version gets initial arguments by [args].
+     *
+     * @param args Arguments used for initial use case initialization.
+     * @return [Result] that encapsulates either a successful result with [Success] or a failed result with [Error]
+     */
+    suspend fun <ARGS, T : Any?> UseCase<ARGS, T>.execute(args: ARGS, cancelPrevious: Boolean = true): Result<T> {
+        if (cancelPrevious) {
+            deferred?.cancel()
+        }
+        return try {
+            val newDeferred = coroutineScope.async(getWorkerDispatcher(), CoroutineStart.LAZY) {
+                build(args)
+            }.also { deferred = it }
+            Success(newDeferred.await())
+        } catch (exception: Throwable) {
+            if (exception is CancellationException) {
```

Comment:
<table data-meta="generated_by_danger-pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/01753e4e83475541b61b01cfce8108bc8990ca63/cr-usecases/src/main/java/com/thefuntasty/mvvm/crusecases/CoroutineScopeOwner.kt#L110"></span>Instead of checking for a general exception type and checking for a specific exception type use multiple catch blocks.</td>
    </tr>
  </tbody>
</table>